libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -533,13 +533,20 @@ bool VoteManager::verifyRewardVote(const std::shared_ptr<Vote>& vote) {
   const auto vote_hash = vote->getHash();
   const auto voted_block_hash = vote->getBlockHash();
   const auto pbft_chain_last_block_hash = pbft_chain_->getLastPbftBlockHash();
+  const auto pbft_chain_last_block_round = pbft_chain_->getLastPbftBlockRound();
 
   if (voted_block_hash != pbft_chain_last_block_hash) {
     LOG(log_dg_) << "Drop vote " << vote_hash << ". PBFT chain last block is " << pbft_chain_last_block_hash
                  << ", but vote for PBFT block is " << voted_block_hash;
     return false;
   }
 
+  if (vote->getRound() != pbft_chain_last_block_round) {
+    LOG(log_dg_) << "Drop vote " << vote_hash << ". PBFT chain last block round is " << pbft_chain_last_block_round
+                 << ", but vote for PBFT block is round " << vote->getRound();
+    return false;
+  }
```

Comment:
and this should be assert as this can not happened or? how we can have different rounds? then it is not cert vote